src/components/cesium_map/api/spatial.js

```diff
@@ -641,17 +643,21 @@ export class ISamplesSpatial {
   gridTracker(viewer, gridder){
     let scratchRectangle = new Cesium.Rectangle();
     let rect = viewer.camera.computeViewRectangle(viewer.scene.globe.ellipsoid, scratchRectangle);
-    if (this.r2str(rect) === this.gridder.global_grid.rect_str){
+    let resultCntChanged = this.prevNumFound !== store.getState()['results']['numFound'];
+    if (this.r2str(rect) === this.gridder.global_grid.rect_str && !resultCntChanged){ // when same boundary and count did not change
```

Comment:
Only update the grid when the bounding box differs and the count of the found points changed.